ksql-common/src/main/java/io/confluent/ksql/schema/ksql/types/SqlType.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.ksql.types;
+
+import com.google.errorprone.annotations.Immutable;
+import io.confluent.ksql.schema.ksql.FormatOptions;
+import io.confluent.ksql.schema.ksql.SqlBaseType;
+import java.util.Objects;
+
+/**
+ * Interface for all SQL types in KSQL.
+ */
+@Immutable
+public abstract class SqlType {
```

Comment:
Note: new base type for our SQL types.  There is no optional/not-optional here, as all types are implicitly optional, (for now).